### DISCUSSION TITLE
Is it possible to determine whether cask is being upgraded?

### DISCUSSION THREAD

**Discussion by Kentzo** [Homebrew/discussions#4996] [Writing Formulae/Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

The Wireshark cask adds a group during installation but deletes this group during uninstallation. Since Homebrew simply uninstall/install during the upgrade, any modifications to the group (such as additional users) get lost.

To avoid this I'd like to let the scripts know via an environment variable that they are run as part of an upgrade, so the group configuration can be preserved.

Is this currently possible?

**Comment by SMillerDev** [comment]
There aren't any environment variables specifically for this purpose. You can check for some of https://docs.brew.sh/Manpage#environment, but no guarantees that all will always be set.

**Reply by Kentzo** [reply]
What is the recommended approach to let cask to alter its installation / uninstallation steps during an upgrade?

**Reply by SMillerDev** [reply]
There is none
